tests/core/test_initialize.py

```diff
@@ -19,709 +17,767 @@
 from conda.common.compat import on_win, open
 from conda.common.io import captured, env_var, env_vars
 from conda.common.path import get_python_short_path, win_path_backout, win_path_ok
-from conda.core.initialize import Result, _get_python_info, init_sh_system, init_sh_user, \
-    initialize_dev, install, install_conda_csh, install_conda_fish, \
-    install_conda_sh, install_conda_xsh, make_entry_point, make_entry_point_exe, \
-    make_initialize_plan, make_install_plan, install_condabin_conda_bat
+from conda.core.initialize import (
+    Result,
+    _get_python_info,
+    init_sh_system,
+    init_sh_user,
+    initialize_dev,
+    install,
+    install_conda_csh,
+    install_conda_fish,
+    install_conda_sh,
+    install_conda_xsh,
+    make_entry_point,
+    make_entry_point_exe,
+    make_initialize_plan,
+    make_install_plan,
+    install_condabin_conda_bat,
+)
 from conda.exceptions import CondaValueError
 from conda.gateways.disk.create import create_link, mkdir_p
 from conda.models.enums import LinkType
 from conda.testing.helpers import tempdir
 
-try:
-    from unittest.mock import Mock, patch
-except ImportError:
-    from unittest.mock import Mock, patch
-
 
 log = getLogger(__name__)
 
 
-class InitializeTests(TestCase):
+@pytest.fixture
```

Comment:
Try adding `fixture(scope="module")` to run this code once, instead of once per test.

Reply:
I wasn't sure if we needed the context to be reset once or per function, it looks like everything is good when I change it

Reply:
I checked the docs, and setUp and tearDown are run after each method. Sorry for the inconvenience.

It does look like some of the test alter context, and would benefit from the `reset_context()` call.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp